pkg/middleware/middleware.go

```diff
@@ -72,6 +87,21 @@ func Process(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, obj cl
 		return dep, nil
 	}
 
+	// If the current object is a PVC, check for volume override
+	if pvc, ok := obj.(*corev1.PersistentVolumeClaim); ok {
+		var volumeSize resource.Quantity
+		switch quayComponentLabel {
+		case "postgres":
+			volumeSize = getVolumeSizeOverrideForComponent(quay, v1.ComponentPostgres)
+		case "clair-postgres":
+			volumeSize = getVolumeSizeOverrideForComponent(quay, v1.ComponentClair)
+		case "objectstorage":
+			volumeSize = getVolumeSizeOverrideForComponent(quay, v1.ComponentObjectStorage)
+		}
+		pvc.Spec.Resources.Requests = corev1.ResourceList{corev1.ResourceStorage: volumeSize}
```

Comment:
What happens when users shrink the storage? Can this operation generate some failure? If yes, does the user see the failure in the quay registry object status?

Reply:
@jonathankingfc ping

Reply:
This is removed from the PR.